타일 관련된 코드 수정

beuoon opened the pull request:
대표적인 코드 수정 이유는 결합도 약화 및 응집도 강화
캡슐화로 코드 중복 제거 및 버그 방지 유도

수정사항
- 팔레트 이름 변경
   식별을 쉽게하기 위해 ~Palette로 변경
- 타일맵 이름 변경 및 용도 지정
   엘리베이터, 장식품 등이 충돌처리 되면 Obstacle과 Object를 분리할 필요가 없을 것으로 생각됨
   (불)가연성 유무의 따라 타일맵을 분리하는 것도 고려해볼만 함
   - Background: 바닥 타일
   - Obstacle: 엘레베이터, 장식품 등
   - Object: 벽, 상호작용 오브젝트 등 (플레이어가 지나가지 못하는 타일 위주)
   - Environment: 불, 물, 전기 등 (생존자 위치 포함 ← 분리할까 생각 중)
   - PlayerSpawn: 플레이어 스폰 위치 (대원 배치 때만 보여줘서 분리)
- 생존자 관련 코드 TileMgr에서 GameMgr로 이동
   생존자는 타일의 느낌이 없어서 GameMgr로 이동했음
- Tile 관련 코드 GameMgr에서 TileMgr로 이동
   이곳저곳에서 Tilemap 가져다가 수정하는게 안좋아보여서 캡술화하는겸 TileMgr로 이동
   캡슐화로 인해 코드 중복이 많이 제거됨(타일 색 수정 등)
   나중에 방식 변경할 때도 해당 코드만 수정하면 되서 용이할 것으로 보임
- Tile Object Generic 삭제
   Tile Object를 Generic으로 갖고 있던 이유가 타일의 Object를 참조하지 못한다고 생각해서였는데, 알고보니 GetInstantiatedObject라는 함수가 있었음
- ~Tile.cs 삭제
   위의 Generic을 삭제하면서 ~Tile.cs에서 하는 일이 아예 없어짐 (기존에는 Generic에 Object 추가하는 일만 했음)
   따라서 기존의 ~Tile.cs 들은 삭제 (Ex. FireTile.cs, WaterTile.cs 등)
   그런데 Tile Asset 만들려면 있긴 있어야 되서 EnvironmentTile과 ObjectTile 만듬 (기능의 차이는 없고, 그냥 이름만 다름)
- Fire, Ember Sprite 변경
   그냥 보기 쉽게하려고 변경 (큰 의미는 없음)
- Ember 생성 방식 변경
   불 개수만큼 작은 불 생성한다고 해서 그냥 불이 작은 불 갖게 만듬
   오브젝트 위치만 변경하는건 기존과 동일하고, 해당 작업은 TileMgr 및 Fire에서 진행
   Embers 삭제
- 불 번짐, 스프링쿨러 미구현
   구조 변경하느라 기존 코드 삭제했는데, 불 번짐 메커니즘 변경되서 미구현으로 냅둠
   스프링쿨러는 구조적 효율성 및 기획에 의문점이 있어서 기획자에게 메커니즘 문의함 (Ex. 스프링쿨러 작동시키면 다른 층의 불도 없어지는지)
- ID Card UI Text 수정은 GameMgr에서 하는걸로 변경
   Player들이 전부 ID Card UI Text를 수정하니까 버그가 있었음 (Ex. 현재 주시 중이 아닌 Player가 피해를 입어도 UI Text 변경)
   버그 방지 및 캡슐화를 위해 GameMgr에서 수정하는 것으로 변경
- RescueTargets 삭제
   특정 위치의 생존자는 GameMgr의 GetRescueTargetAt 사용
   전체 사용자 리스트는 GameObject.Find 사용

beuoon committed:
Fix: GameMgr의 Comp_Players 접근 제한자 수정

결합도 약화를 위해 GameMgr의 Comp_players 접근 제한자 강화

beuoon committed:
Fix: 타일 관련된 코드 대수정

beuoon committed:
Fix: RescueTargets, Embers 삭제

불필요한 오브젝트 삭제

Myoryung commented:
문제가 있습니다. 이번에 우선순위에 캐릭터 세모부분을 수정하면서 지금 playerAct같은 부분이 구현부분이랑 겹치는데 merge할때 어떻게 수정해야 합니까 ㅋㅋㅋ

beuoon committed:
Add: 도구 시각 개선, 캡틴 스킬 구현

도구 사용 시 어디에 사용되는지 보이지 않아서 시각적인 부분 개선
도구 사용 중 이동시 끊김
도구 선택해도 UI 안꺼지는 버그 수정

캡틴 스킬이 모든 턴의 재난이 보이던 버그 수정
캡틴 스킬과 이펙트가 겹칠 때 우선순위를 나누기 위해서 Warning Tilemap과 Effect Tilemap 추가
